sdks/java/io/solace/src/test/java/org/apache/beam/sdk/io/solace/it/SolaceContainerManager.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.solace.it;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+import org.testcontainers.solace.Service;
+import org.testcontainers.solace.SolaceContainer;
+import org.testcontainers.utility.DockerImageName;
+
+public class SolaceContainerManager {
+
+  public static final String VPN_NAME = "default";
+  public static final String PASSWORD = "password";
+  public static final String USERNAME = "username";
+  public static final String TOPIC_NAME = "test_topic";
+  private static final Logger LOG = LoggerFactory.getLogger(SolaceContainerManager.class);
+  private final SolaceContainer container;
+
+  public SolaceContainerManager() {
+    this.container =
+        new SolaceContainer(DockerImageName.parse("solace/solace-pubsub-standard:10.7")) {
+          {
+            addFixedExposedPort(55555, 55555);
```

Comment:
the available port on test VM is unpredictable. If these ports are occupied the test could fail. Is this a concern? Or do something similar to https://github.com/apache/beam/blob/8a4a250076bde59fadb89b2e580e6e98d064b26d/sdks/java/core/src/main/java/org/apache/beam/sdk/util/construction/TransformUpgrader.java#L400

Reply:
That's a good point, thanks. I've implemented the findAvailablePort() function and removed unnecessary port exposures.